### ISSUE TITLE
Include Haskell packages documentation only if documentation.doc.enable is set

### ISSUE THREAD

**Issue by iblech** [NixOS/nixpkgs#316768]
From the description of the `documentation.doc.enable` option, I would have expected that setting it to `false` inhibits the installation of most docs.

However, `ghcWithPackages` seems to unconditionally pull documentation in (almost 700 MiB on my system).

I believe the culprit is `installDocumentation ? true` at https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/haskell-modules/with-packages-wrapper.nix#L8. I'd be happy to contribute a pull request but would require some guidance.

**Comment by cdepillabout** [comment]
@iblech Could you explain more what your local setup looks like?

`documentation.doc.enable` (I think) is a NixOS option, while `ghcWithPackages` is a function within Nixpkgs.  They are not related at all.  Calls to `ghcWithPackages` won't be able to see your NixOS options.

**Comment by iphydf** [comment]
I'd expect the following to work:

`ghc = haskellPackages.ghcWithPackages.override { installDocumentation = false; } (p: with p; [ ... ]);`

But it doesn't seem to. I'm quite confused about why. https://haskell4nix.readthedocs.io/frequently-asked-questions.html would suggest that this should work.

**Comment by acid-bong** [comment]
Ight, i wanted to file a ticket about the GHC wrapper and just realized your issue is already about it.

A couple of words on NixOS: `buildEnv`, which builds the system path, only accepts extra outputs to link, it doesn't filter out an output not supplied in `extraOutputsToInstall`. By default, NixOS links "man", "doc" and "info", but it doesn't filter out the rest. You always can explicitly install a `PACKAGE.doc` even with `documentation.doc.enable = false` (which adds "doc" to `environment.extraOutputsToInstall`).

TLDR: your problem is that `ghc.withPackages` always installs documentation

Possible solution: don't install compilers/interpreters globally, keep them in devshells only
